src/Blockcore.Indexer.Core/Storage/Mongo/BlockRewindOperation.cs

```diff
@@ -61,18 +65,44 @@ public static async Task RewindBlockAsync(this MongoData storage, long blockInde
       Task<DeleteResult> addressHistoryComputed =
          storage.AddressHistoryComputedTable.DeleteManyAsync(addrCompHistFilter);
 
+      await Task.WhenAll( output, transactions, addressComputed, addressHistoryComputed);
+
+      await MergeRewindInputsToUnspentTransactionsAsync(storage, blockIndex);
```

Comment:
So the order here is for the given block say block x that is reorged
- merge from `InputTable` all the deleted outputs by block X back to the `UnspentOutputTable`
- delete all the inputs of block X from the `InputTable`
- delete all the utxo added by block X from the `UnspentOutputTable`